container/lxd/server.go

```diff
@@ -158,6 +158,53 @@ func (s *Server) UpdateContainerConfig(name string, cfg map[string]string) error
 	return errors.Trace(resp.Wait())
 }
 
+// ReplaceOrAddContainerProfile updates the profiles for the container with the
+// input name, using the input values.
+func (s *Server) ReplaceOrAddContainerProfile(name, oldProfile, newProfile string) error {
+	container, eTag, err := s.GetContainer(name)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	profiles := container.Profiles
+	if oldProfile == "" {
+		// add profile
+		profiles = append(profiles, newProfile)
+	} else {
+		for i, pName := range profiles {
+			if pName == oldProfile {
+				if newProfile == "" {
+					// remove profile
+					profiles = append(profiles[:i], profiles[i+1:]...)
+				} else {
+					// replace profile
+					profiles[i] = newProfile
+				}
+				break
+			}
+		}
+	}
```

Comment:
It feels like a diff/union type of situation going on. I wonder if we can use `set.Strings`, which provide this to some degree?

Reply:
My concern is that set.Strings doesn't have a Replace() it has Remove() and Add() which is good.   Am I missing a trick?

Right now, the profile upgrade leaves the profiles in order by application.  If we want the upgrade to put the profile at the end instead, I can definitely change to using set.Strings.

Reply:
per conversation, i'll pull this out into it's own function for easier testing.

Reply:
done